### ISSUE TITLE
move storage type conversion trait from ImageCore to Colors

### ISSUE THREAD

**Issue by johnnychen94** [JuliaGraphics/Colors.jl#475]
It looks like the most appropriate place for these traits is here:

https://github.com/JuliaImages/ImageCore.jl/blob/4fb132187d68054be7413ec229071115c14f72bd/src/convert_reinterpret.jl#L61-L107

**Comment by kimikage** [comment]
Also, as you know, there is the inconsistency problem with `float` other than `FixedPoint`.

**Comment by johnnychen94** [comment]
> I have no objection to "defining" them in Colors or ColorTypes, but I am not comfortable exporting them.

I'm okay with this since it's mainly the package developer that needs to take care of the storage type.

My main consideration when proposing this move is that I want to give better support for them on broader color types. Currently, 

```julia
julia> n0f8(RGB)
RGB{N0f8}

julia> n0f8(RGB{Float32})
RGB{N0f8}

julia> n0f8(RGB24)
ERROR: TypeError: in Type{...} expression, expected UnionAll, got Type{RGB24}
Stacktrace:
 [1] n0f8(#unused#::Type{RGB24})
   @ ImageCore ~/.julia/packages/ImageCore/iXG0W/src/convert_reinterpret.jl:82
 [2] top-level scope
   @ REPL[21]:1
```

This need arises when I tried to convert all colorant images to N0f8/UInt8 byte sequences (in my WIP package [Sixel.jl](https://github.com/johnnychen94/Sixel.jl/pull/1/files#diff-f2c4483d5a03d01006c0be9398373f415c4fe752eb31a0028bf5055c203aca42R79-R81))

> but I don't see what the trouble is with having them in ImageCore.

The only reason that I feel it better to live in Colors is that they are pixel-level operations. 😄

**Comment by kimikage** [comment]
Even if we do the migration, I don't see any reason not to fix `n0f8(RGB24)` etc. in the current ImageCore. (I'm not sure whether `n0f8(RGB24)` should return `RGB{N0f8}` or `RGB24`, though.)

**Comment by kimikage** [comment]
I think it's straightforward to define `n0f8` and so on in `FixedPointNumbers` and implement `n0f8(::Colorant)` and so on in `ColorTypes`. However, I'm not sure that it's really a good idea for `FixedPointNumbers` to provide such non-generic functions.
Also, I don't think `float32` and `float64` are functions that should be exported by the low-level packages.

However, I think the following should be solved within ColorTypes.
```julia
julia> convert(Colorant{Float32}, RGB24(1))
ERROR: nonparametric type RGB24 has ambiguous destination Colorant{Float32, N} where N
```
(Of course, this is a known issue:
https://github.com/JuliaGraphics/ColorTypes.jl/blob/81297fff41e28dc64ace72a37c090ee22363ba71/test/conversions.jl#L446)

**Comment by johnnychen94** [comment]
Makes sense to me; either move or not are good. We can keep this issue open until we decide the 1.0 status.
